tests/data/test_transformer.py

```diff
@@ -0,0 +1,69 @@
+import os
+import shutil
+from pathlib import Path
+
+import numpy as np
+import pytest
+import torch
+from sklearn.model_selection import train_test_split
+
+from birdclef.birdnet import load_model
+from birdclef.data.datasets import AudioPCMDataSet
+from birdclef.data.transforms import ToBirdNETEmbedding
+from birdclef.transformer import TransformerModel, create_mask, split_and_copy_files
+
+
+def test_create_mask():
+    embeddings = torch.rand(32, 320)
+    mask = create_mask(embeddings)
+    assert mask.size() == embeddings.size()
+    assert torch.sum(mask).item() > 0
+
+
+def test_TransformerModel():
+    input_dim = 320
+    model_dim = 256
+    num_heads = 8
+    num_layers = 4
+    model = TransformerModel(input_dim, model_dim, num_heads, num_layers)
+
+    x = torch.rand(32, 320)
+    y = model(x)
+
+    assert y.size() == x.size()
+
+
+def test_split_and_copy_files(tmp_path):
+    parent_dir = Path(__file__).parent.parent
+    source_dir = parent_dir / "data/raw/birdclef-2022/train_audio/mp3/"
```

Comment:
We can't rely on the data directory being here. Try using this test fixture for split_and_copy_files: https://github.com/dsgt-birdclef/birdclef-2023/blob/main/tests/conftest.py#L14